sdk/identity/Azure.Identity/src/CloudShellManagedIdentitySource.cs

```diff
@@ -36,38 +35,31 @@ public static ManagedIdentitySource TryCreate(ManagedIdentityClientOptions optio
                 throw new AuthenticationFailedException(MsiEndpointInvalidUriError, ex);
             }
 
-            return new CloudShellManagedIdentitySource(options.Pipeline, endpointUri, options.ClientId);
+            return new CloudShellManagedIdentitySource(endpointUri, options);
         }
 
-        private CloudShellManagedIdentitySource(CredentialPipeline pipeline, Uri endpoint, string clientId) : base(pipeline)
+        private CloudShellManagedIdentitySource(Uri endpoint, ManagedIdentityClientOptions options) : base(options.Pipeline)
         {
             _endpoint = endpoint;
-            _clientId = clientId;
+            if (!string.IsNullOrEmpty(options.ClientId) || null == options.ResourceIdentifier)
+            {
+                AzureIdentityEventSource.Singleton.UserAssignedManagedIdentityNotSupported("CloudShell");
```

Comment:
Should this be the product name?
```suggestion
                AzureIdentityEventSource.Singleton.UserAssignedManagedIdentityNotSupported("Cloud Shell");
```